### PR TITLE
fix!: update unbonding  pausing for PSS

### PR DESCRIPTION
**Pull request by sainoe**
Note that if this solution is adopted it will require some migrations.

Closes: #1727

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] Added `!` to the type prefix if the change is [state-machine breaking](https://github.com/cosmos/interchain-security/blob/main/RELEASES.md#breaking-changes)
* [ ] Confirmed this PR does not introduce changes requiring state migrations, OR migration code has been added to consumer and/or provider modules
* [ ] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] Provided a link to the relevant issue or specification
* [ ] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/00-intro.md)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
* [ ] Added a changelog entry to `CHANGELOG.md`
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [ ] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed
* [ ] If this PR is library API breaking, bump the go.mod version string of the repo, and follow through on a new major release

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` the type prefix if the change is state-machine breaking
* [ ] confirmed this PR does not introduce changes requiring state migrations, OR confirmed migration code has been added to consumer and/or provider modules
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: x/ccv/provider/keeper/hooks.go [hunk 1]
```diff
@@ -35,8 +36,63 @@ func (k *Keeper) Hooks() Hooks {
 func (h Hooks) AfterUnbondingInitiated(ctx sdk.Context, id uint64) error {
 	var consumerChainIDS []string
 
+	// get validator address from unbonding operation
+	unbondingType, found := h.k.stakingKeeper.GetUnbondingType(ctx, id)
+	vadAddrBech32 := ""
+	if !found {
+		ctx.Logger().Error("undefined type for unbonding operation id: %d", id)
```
Comment (insumity) [inline x/ccv/provider/keeper/hooks.go:43]: Should this and the ones below return an actual error?

File: x/ccv/provider/keeper/hooks.go [hunk 2]
```diff
@@ -35,8 +36,63 @@ func (k *Keeper) Hooks() Hooks {
...
+		vadAddrBech32 = red.ValidatorSrcAddress
+	case stakingtypes.UnbondingType_ValidatorUnbonding:
+		val, found := h.k.stakingKeeper.GetValidatorByUnbondingID(ctx, id)
+		if !found {
+			ctx.Logger().Error("unfound validator for unbonding operation id: %d", id)
+			return nil
+		}
+		vadAddrBech32 = val.OperatorAddress
+	default:
+		panic(stakingtypes.ErrUnbondingNotFound)
```
Comment (p-offtermatt) [inline x/ccv/provider/keeper/hooks.go:70]: Just a reminder, you mentioned the panic here should be removed :)

**Review by p-offtermatt — APPROVED** [review]
LGTM. 
As mentioned, this leaves some extreme edge cases where unbondings that should be paused do not get unbonding-paused, but seems ok as a fix for now.

File: x/ccv/provider/keeper/hooks.go (hunk 1 above)
Comment (sainoe) [reply x/ccv/provider/keeper/hooks.go:43]: Returning an error here can potentially halt the provider when the hook is called after a validator starts unbonding in [val_state_change.go](https://github.com/cosmos/cosmos-sdk/blob/173f2ea773b47aaa99f878a58fdb6e09f0fbd717/x/staking/keeper/val_state_change.go#L354). So my suggestion is to stick to just logging those errors from the hook's perspective without returning any.

Comment (insumity) [reply x/ccv/provider/keeper/hooks.go:43]: That makes sense. Thanks!

**Review by insumity — APPROVED** [review]

**Pull request merged**